alphastats/DataSet_Preprocess.py

```diff
@@ -9,16 +10,55 @@
 import streamlit as st
 
 from sklearn.experimental import enable_iterative_imputer  # noqa
-from tqdm import tqdm
 
 from alphastats.utils import ignore_warning
 
 
-class Preprocess:
+class PreprocessInterface(ABC):
```

Comment:
Does this have any added benefit besides error catching by the IDE and linters?

To me it looks like it mostly increases complexity and duplicates the definition of the function arguments.

If we use this, I think it should also define, which shared attributes should be present in the child classes, so we have all the documentation of the interface in one place.

Reply:
I would leave it like this for now, "error catching by the IDE and linters" is alreday quite some value ..also it makes the connection between the implementation and the interface transparent.. 
The shared attributes are already defined: there are none ;-)

Reply:
update: I realized these are two difference methods .. remove the interface for now